### PR TITLE
Add wrappers for common operation signatures to rtt_std_srvs

### PR DESCRIPTION
**Pull request by meyerj**
Implements #101.

With this patch it is possible to create ROS service servers for some common operation signatures out of the box, without the need to write custom wrappers or to add extra operations with the ROS specific callback signature.

### Supported service types and associated signatures:
```
std_srvs/Empty:
- bool empty()                     // The service call fails if empty() returns false!
                                   // Use std_srvs/Trigger if the result should be returned as the response.
- void empty()

std_srvs/SetBool:
- bool setBool(bool, std::string &message_out)
- bool setBool(bool)               // response.message will be empty
- std::string setBool(bool)        // response.success = true
- void setBool(bool)               // response.success = true and response.message will be empty

std_srvs/Trigger:
- bool trigger(std::string &message_out)
- bool trigger()                   // response.message will be empty
- std::string trigger()            // response.success = true
```

The approach can be easily extended to other ROS service types.

### CODE REVIEW

File: tests/rtt_roscomm_tests/test/transport_tests.cpp [hunk 1]
```diff
@@ -118,12 +118,23 @@ TEST(TransportTest, VectorTest)
 }
 
 static int callback_called = 0;
```
Comment (francisco-miguel-almeida) [inline tests/rtt_roscomm_tests/test/transport_tests.cpp:120]: question: do we control in which thread the callbacks are made? If not, maybe this counter should be atomic. Not important right now, just curious.

**Review by francisco-miguel-almeida — APPROVED** [review]
LGTM, there was just a build failure due to pre-C++11 incompatibility. Issue is easily circumvented, however.

File: tests/rtt_roscomm_tests/test/transport_tests.cpp (hunk 1 above)
Comment (meyerj) [reply tests/rtt_roscomm_tests/test/transport_tests.cpp:120]: The callbacks are executed by a thread pool owned by a `ros::AsyncSpinner` in `rtt_rosnode`. They could execute in parallel, but the service calls in the unit tests are synchronous and block until the callback has been processed and the results are returned through the network stack and roscpp. So no concurrency.

**Comment by francisco-miguel-almeida** [comment]
> LGTM, there was just a build failure due to pre-C++11 incompatibility. Issue is easily circumvented, however.

Was not quite as easily circumvented as it appeared at first, but now it works as intended.
We can address the SFINAE hackiness later.

**Comment by ahoarau** [comment]
Very nice !

**Comment by ahoarau** [comment]
very nice !

**Pull request merged**